app/models/manageiq/providers/azure/cloud_manager/refresh_parser.rb

```diff
@@ -124,6 +135,49 @@ def get_images
         process_collection(images, :vms) { |image| parse_image(image) }
       end
 
+      def get_security_groups
+        security_groups = gather_data_for_this_region(@nsg)
+
+        process_collection(security_groups, :security_groups) do |sg|
+          parse_security_group(sg)
+        end
+
+        get_firewall_rules
+      end
+
+      # To get the list of security groups for a VM we need to find any
+      # NIC's associated with the VM, and then get the security groups
+      # associated with the NIC's.
+      #
+      def get_security_groups_for_vm(vm)
+        interfaces = vm.properties.network_profile.network_interfaces.map do |nic|
+          File.basename(nic.id)
+        end
+
+        # A nic does not necessarily have an associated network security group.
+        security_group_list = interfaces.map do |nic|
+          nic_instance = @nis.get(nic, vm.resource_group)
+          ns_group = nic_instance.properties['networkSecurityGroup']
+          ns_group.id if ns_group
+        end
+
+        # The resource group of the nic or vm is not necessarily the same as the security group.
+        security_groups = security_group_list.compact.map do |sg|
+          security_group_name = File.basename(sg)
+          resource_group = sg[%r{resourceGroups/(.*?)/}, 1]
+          @nsg.get(security_group_name, resource_group)
```

Comment:
@bronaghs True. I think I could invert this so that we go from security group to VM instead of the other way around. We still need the call to get NIC's, but it would be one less call overall.